### PR TITLE
feat(playback): add random variation on each keypress (#24)

### LINKED ISSUE #24: Distinct sound for keys
**Is your feature request related to a problem? Please describe.**

For the default preset, having a distinct sound for each key press would be an interesting feature. Maybe it might even sound more realistic.

**Describe the solution you'd like**

n/a
Ideas welcome.

**Describe alternatives you've considered**

n/a

**Additional context**

https://www.reddit.com/r/linux/comments/16xu9ek/comment/k37m3qj/?utm_source=share&utm_medium=web2x&context=3

### PR DESCRIPTION
**Pull request by Xenira**
## Description of change

Adds the possibility of random variation for each keypress
- [x]  Volume
- [x]  Tempo / Pitch

I added some values to the default preset. Maybe play around with those to find sensible defaults. :heavy_check_mark: 

Closes #24 

### Configuration
I added configuration possibilities in 3 places:

1. CLI Arguments. These will override all settings defined in Presets.
2. `KeyConfig` in Preset. These will override settings defined in the parent Preset.
3. Preset. Will apply to all `KeyConfig`s that do not provide variation.

~~Currently the cli only accepts one value, that is used for both changes up and down, while the config file takes separate values. I am looking into a convenient way to allow both for the cli as well.~~

The cli accepts (up to) two comma separated values. If only one is defined it is used for both.

## How has this been tested? (if applicable)

- Set high values in config and use :ear: 
- Set high values in cli and use :ear: 
- Watch trace logs.

### CODE REVIEW

File: src/args.rs [hunk 1]
```diff
@@ -52,6 +54,36 @@ pub struct Args {
...
+    /// Variate pitch/volume/tempo.
+    #[command(flatten)]
+    pub sound_variation_args: Option<SoundVariationArgs>,
+}
+
+/// Variate pitch/volume/tempo.
+#[derive(clap::Args, Default, Debug)]
+pub struct SoundVariationArgs {
+    /// Variate pitch +/- in percent.
+    /// Overrides preset configuration.
```
Comment (orhun) [inline src/args.rs:66]: Instead of mentioning this here, maybe we can add it to the docs.

**Review by orhun — COMMENTED** [review]
The code looks pretty good! I tried it out and works as expected. Also, left a comment about changing the pitch in #24 

Feel free to rebase `main` and update docs when you have time. 

cc @arda-guler for testing / finding the most sane defaults.

**Comment by arda-guler** [comment]
Yeah, well, regarding that "finding the most sane defaults" thing and the pitch implementation...

The indirect implementation here by changing the speed is more than adequate enough IMO. In fact, I'm going to say that as far as my own opinion goes, a 0.1 volume variation and a 0.05 tempo variation sounds so much better - the current example values are a bit too pronounced. It should be a little more subtle.

All else is very good!

**Comment by Xenira** [comment]
Sorry for the git history confusion... Forgot to sign and messed up fixing it...

**Review by orhun — APPROVED** [review]
Great stuff!

**Pull request merged**